gateway/src/main/java/io/camunda/zeebe/gateway/impl/job/LongPollingActivateJobsHandler.java

```diff
@@ -66,15 +70,23 @@ private LongPollingActivateJobsHandler(
     metrics = new LongPollingMetrics();
   }
 
-  @Override
   public String getName() {
     return "GatewayLongPollingJobHandler";
   }
 
   @Override
+  public void accept(ActorControl actor) {
+    this.actor = actor;
+    onActorStarted();
+  }
+
   protected void onActorStarted() {
-    brokerClient.subscribeJobAvailableNotification(JOBS_AVAILABLE_TOPIC, this::onNotification);
-    actor.runAtFixedRate(Duration.ofMillis(probeTimeoutMillis), this::probe);
+    actor.run(
+        () -> {
+          brokerClient.subscribeJobAvailableNotification(
+              JOBS_AVAILABLE_TOPIC, this::onNotification);
+          actor.runAtFixedRate(Duration.ofMillis(probeTimeoutMillis), this::probe);
+        });
```

Comment:
> It's not obvious to me, why this is the only code block that is wrapped into an actor job.

Just to ensure that we have the same understanding of why the code block must be wrapped into an actor job: Calling `#runAtFixedRate()` checks if it is called "from within actor":

https://github.com/camunda-cloud/zeebe/blob/2027d53cb71848be659dd66ffa0f90d87e1a6fbb/util/src/main/java/io/camunda/zeebe/util/sched/ActorControl.java#L400-L414

To be more concrete, before that change, the class `LongPollingActivateJobsHandler` extended the `Actor` class. By inheriting from `Actor`, the method `#onActorStarted()` was called during the bootstrap phase of the `Actor`. In that case, it was safe to assume that `onActorStarted()` was called "from within" the correct actor, so that calling `#runAtFixedRate()` would schedule a timer subscription by using the correct actor.

With changing the implementation to delegation (instead of inheritance) to the actor, the assumption may not apply anymore. So, there is no guarantee anymore that `#onActorStarted()`, and hence `#runAtFixedRate()`, is called "from within" the correct actor. It could be but it doesn't need to be. Basically, the method `#accept(final ActorControl actor)` may be called by a different thread and/or actor that passes the "correct" actor, and in that case, if the code block wouldn't be wrapped into an actor job, the execution of `#runAtFixedRate()` will fail.

To ensure that `#runAtFixedRate()` is called "from within" the correct actor and that the timer subscription is scheduled at the correct actor, there is a need to wrap that code block to an actor job.

> Why not wrap the other method bodies into actor jobs as well?

As the `LongPollingActivateJobsHandler` was already interacting with the `actor` prior to the changes, its implementation already ensures that all other critical code blocks are submitted as actor jobs whenever it is required.

Hope that helps to get a better understanding?